scripts/skills/perks/perk_rf_whirling_death.nut

```diff
@@ -1,46 +1,168 @@
 this.perk_rf_whirling_death <- ::inherit("scripts/skills/skill", {
 	m = {
-		IsForceEnabled = false,
-		MinBonus = 0,
-		MaxBonus = 25
+		RequiredWeaponType = ::Const.Items.WeaponType.Flail,
+		Stacks = 0,
+		IsPerformingExtraAttack = false, // Is flipped during the extra attack to reduce its damage using onAnySkillUsed
+		DamageTotalMult = 0.5, // Multiplier for the damage dealt during the extra attack
+		ChanceToHitHeadPerReachDiff = 10 // Is used for two-handed flails
 	},
 	function create()
 	{
 		this.m.ID = "perk.rf_whirling_death";
 		this.m.Name = ::Const.Strings.PerkName.RF_WhirlingDeath;
-		this.m.Description = ::Const.Strings.PerkDescription.RF_WhirlingDeath;
+		this.m.Description = "This character\'s attacks are like a whirlwind, making it very dangerous to be near them.";
 		this.m.Icon = "ui/perks/rf_whirling_death.png";
-		this.m.Type = ::Const.SkillType.Perk;
+		this.m.Type = ::Const.SkillType.Perk | ::Const.SkillType.StatusEffect;
 		this.m.Order = ::Const.SkillOrder.Perk;
-		this.m.IsActive = false;
-		this.m.IsStacking = false;
-		this.m.IsHidden = false;
 	}
 
-	function isEnabled()
+	function isHidden()
 	{
-		if (this.m.IsForceEnabled)
+		return this.m.Stacks == 0;
+	}
+
+	function getTooltip()
+	{
+		local ret = this.skill.getTooltip();
+		ret.push({
+			id = 10,
+			type = "text",
+			icon = "ui/icons/reach.png",
+			text = ::Reforged.Mod.Tooltips.parseString(::MSU.Text.colorizeValue(this.m.Stacks) + " [Reach|Concept.Reach]")
+		});
+		ret.push({
+			id = 20,
+			type = "text",
+			icon = "ui/tooltips/warning.png",
+			text = "Will expire upon swapping your weapon!"
+		});
+		return ret;
+	}
+
+	function onUpdate( _properties )
+	{
+		_properties.Reach += this.m.Stacks;
+	}
+
+	function onAnySkillUsed( _skill, _targetEntity, _properties )
+	{
+		if (this.m.IsPerformingExtraAttack)
+			_properties.DamageTotalMult *= this.m.DamageTotalMult;
```

Comment:
How does this work when **Flail Spinner** triggers during this extra attack?
My gut feeling says that **Flail Spinner**'s attack would be nerfed twice down to 25% damage because it happens deep within the extra attack from this perk, so `this.m.IsPerformingExtraAttack` is never set to `false` in between.

From the description one would think that **only** the one(1) singular attack caused by this perk would be nerfed and not any child attacks spawned by other things as a reaction to that.

Reply:
Basically when the target or attacker is visible to the player i.e. we are using the `::Time.scheduleEvent` methodology in both this perk and Flail Spinner, then both the perks act independently of each other because their variables are set/reset during that scheduled event.

However, when the target and attacker both are **not visible to the player**, then both the variables will be applied as then we aren't using scheduled events (this is following vanilla code style probably for performance and/or other hidden reasons). So this is definitely a situation that we should try to fix.

Reply:
I've pushed a fix for this in a new commit. Please check if it looks ok to you.